src/library_vr.js

```diff
@@ -0,0 +1,227 @@
+var LibraryWebVR = {
+  $WebVR: {
+    DEVICE_TYPE_UNKNOWN: 0,
+    DEVICE_TYPE_HMD: 1,
+    DEVICE_TYPE_SENSOR: 2,
+
+    EYE_LEFT: 0,
+    EYE_RIGHT: 1,
+
+    initialized: false,
+    ready: false,
+    devices: [],
+    deviceHardwareIds: {},
+    nextHardwareDeviceId: 1,
+    getDevicesPromise: null,
+
+    selectedHMD: null,
+    selectedHMDId: 0,
+
+    init: function() {
+      if (WebVR.initialized)
+        return true;
+
+      WebVR.initialized = true;
+
+      if (!navigator.getVRDevices) {
+        WebVR.ready = true;
+        WebVR.devices = [];
+        return true;
+      }
+
+      navigator.getVRDevices().then(
+        function(devs) {
+          WebVR.ready = true;
+          WebVR.devices = devs;
+        },
+        function() {
+          WebVR.ready = true;
+          WebVR.devices = [];
+        }
+      );
+    },
+
+    getDeviceByID: function(deviceId) {
+      if (deviceId < 1 || deviceId > WebVR.devices.length) {
+        console.log("library_vr getDeviceByID error: " + (new Error()).stack);
+        throw "Invalid VR device ID given to Emscripten WebVR API function!";
```

Comment:
Throwing here will be uncatchable in C code, so passing an invalid ID to any of the C functions `emscripten_vr_get_device_hwid` and so on below will abort execution without code being able to detect and handle errors gracefully.